### PR TITLE
Fix treasury's tests for v2.0.0-alpha.3

### PR DESCRIPTION
**Pull request by clearloop**
#325 

---

## Steps

+ [x] Migrate new tests from v2.0.0-alpha.3
+ [x] Fix tests

### CODE REVIEW

File: frame/treasury/src/mock.rs [hunk 1]
```diff
@@ -116,12 +116,12 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
...
-		balances: vec![(0, 100), (1, 98), (2, 1)],
-	}
-	.assimilate_storage(&mut t)
-	.unwrap();
+	// pallet_kton::GenesisConfig::<Test> {
+	// 	// Total issuance will be 200 with treasury account initialized at ED.
+	// 	balances: vec![(0, 100), (1, 98), (2, 1)],
+	// }
+	// .assimilate_storage(&mut t)
+	// .unwrap();
```
Comment (clearloop) [inline frame/treasury/src/mock.rs]: KTON's GenesisConfig will override the RING's, haven't figured out why yet.

Comment (AurevoirXavier) [reply frame/treasury/src/mock.rs]: I'll make a pr to fix this

**Comment by clearloop** [comment]
Should support `tip` for KTON? If so, it might be an ugly support...or we support it after we merge KTON and RING?

**Comment by AurevoirXavier** [comment]
> Should we support `tip` for KTON? If so, it might be an ugly support...or we support it after we merge KTON and RING?

The merge means avoid the dup code, they still there. You probably still need two types to acess the different storage.

**Comment by HackFisher** [comment]
> Should we support `tip` for KTON? If so, it might be quite ugly currently now...or we support it after we merge KTON and RING?
> 
> #320

Let's do it later or in a separate issue.

**Comment by AurevoirXavier** [comment]
> > Should we support `tip` for KTON? If so, it might be quite ugly currently now...or we support it after we merge KTON and RING?
> > #320
> 
> Let's do it later or in a separate issue.

Yes.

And I think the staking tests (#300) are the highest priority.
You can pick it up and work with @yanganto .

**Comment by clearloop** [comment]
Ok, I'll join #300 after merging this pr.

**Pull request merged**